### PR TITLE
Updated BCD for Scroll-Driven Animations spec

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

The [Scroll-driven animations spec](https://drafts.csswg.org/scroll-animations-1/) is being more or less supported in its entirety in Chromium 115. This PR adds BCD for the new properties, functions, and other syntax defined in this spec, plus some related features from the [CSS Animations Level 2](https://drafts.csswg.org/css-animations-2/).

See my [research document](https://docs.google.com/document/d/1GC7uqhFl2CIyXhLXvb3ODvwnd_sEx9BPNLk7WrIuJFI/edit#) for more details as to what has been added.

### CODE REVIEW

**Comment by Elchi3** [comment]
So sorry to caused a merge conflict here. I forgot about this PR and merged https://github.com/mdn/browser-compat-data/pull/19977. I will take a look at this tomorrow.

File: css/properties/scroll-timeline-axis.json [hunk 1]
```diff
@@ -7,6 +7,10 @@
           "spec_url": "https://drafts.csswg.org/scroll-animations/#propdef-scroll-timeline-axis",
           "support": {
             "chrome": [
+              {
+                "version_added": "115",
+                "notes": "Supports the <code>x</code> and <code>y</code> values, and not the deprecated <code>horizontal</code> and <code>vertical</code> values."
```
Comment (Elchi3) [inline css/properties/scroll-timeline-axis.json:12]: I'd leave this out. Only have a note when things are not according to the standard. Like in the case of Firefox.

```suggestion
                "version_added": "115"
```

File: css/properties/view-timeline-axis.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  "css": {
+    "properties": {
+      "view-timeline-axis": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/view-timeline-axis",
+          "spec_url": "https://w3c.github.io/csswg-drafts/scroll-animations-1/#view-timeline-axis",
+          "support": {
+            "chrome": {
+              "version_added": "115",
+              "notes": "Supports the <code>x</code> and <code>y</code> values, and not the deprecated <code>horizontal</code> and <code>vertical</code> values."
```
Comment (Elchi3) [inline css/properties/view-timeline-axis.json:11]: Same here.
```suggestion
              "version_added": "115"
```

File: browsers/chrome.json [hunk 3]
```diff
@@ -793,13 +793,13 @@
           "engine_version": "113"
         },
         "114": {
-          "release_date": "2023-05-30",
+          "release_date": "2023-05-24",
```
Comment (Elchi3) [inline browsers/chrome.json:796]: Please separate out this change and the change to chrome_android.json to a new PR.

**Review by Elchi3 — COMMENTED** [review]
Well done! Almost ready to merge. Your homework from me:

- One minor suggestion for the notes. 
- Separating out the release dates to a different PR
- Resolving merge conflicts

**Comment by chrisdavidmills** [comment]
@Elchi3 changes made as requested, and we now have a passing build. I'll submit a separate PR for the browser changes in a moment.

**Review by Elchi3 — APPROVED** [review]
Thank you! 👍

File: css/properties/animation.json [hunk 4]
```diff
@@ -106,6 +106,41 @@
...
           }
+        },
+        "animation-timeline": {
+          "__compat": {
+            "description": "<code>animation-timeline</code> included in shorthand",
+            "support": {
+              "chrome": {
+                "version_added": "115",
+                "partial_implementation": true,
+                "notes": "Support is currently reset-only. Including <code>animation</code> resets a previously-declared <code>animation-timeline</code> value to <code>auto</code>, but <code>animation-timeline</code> cannot be set via <code>animation</code>."
```
Comment (foolip) [inline css/properties/animation.json:117]: This came up in web-features, and I've filed https://crbug.com/337580010 to track this bug, and get clarity on whether it is a bug.

**Pull request merged**